src/binary-reader-linker.cc

```diff
@@ -243,11 +261,16 @@ Result BinaryReaderLinker::OnExport(Index index,
                                     ExternalKind kind,
                                     Index item_index,
                                     StringSlice name) {
+  if (kind == ExternalKind::Memory) {
+    WABT_FATAL("Linker does not support exported memories");
+  }
+
```

Comment:
I'm kinda tempted to say that wasm-link should only work on a certain subset of inputs that satisfy certain constrains.    I makes sense to me if its the linker that decides if memory is imported or exported and how it is finally layed out, not something the inputs the linker should be dictating.

However, I imagine that you have certain modules that you want to link that come from other tools.   s2wasm perhaps?  Would it makes sense to instead have those tools output "linkable" / "relocatable" objects?   How were your input objects generated?  Do they contain relocations?